api/v1alpha1/gatewayclass_types.go

```diff
@@ -61,6 +61,29 @@ type GatewayClassSpec struct {
 	// +required
 	Controller string `json:"controller" protobuf:"bytes,1,opt,name=controller"`
 
+	// GatewayNamespaceSelector is a selector of valid namespaces that Gateways
+	// can use this GatewayClass from. A nil or empty selector (default)
+	// indicates that Gateways can use this GatewayClass from any namespace.
+	//
+	// Support: Core
+	//
+	// +optional
+	GatewayNamespaceSelector *metav1.LabelSelector `json:"gatewayNamespaceSelector" protobuf:"bytes,2,opt,name=gatewayNamespaceSelector"`
+
+	// RouteNamespaceSelector is a selector of valid namespaces that Gateways of
+	// this class can reference Routes in. This allows admins to limit the
+	// namespaces that Gateways of certain classes can reference. This field is
+	// focused on authorization while the similar Route namespace selector on
+	// the Gateway is focused on implementation. A nil selector (default)
+	// indicates that Gateways of this class can reference Routes within the
+	// same namespace. An empty selector indicates that Gateways can reference
+	// Routes in any namespace.
+	//
+	// Support: Core
+	//
+	// +optional
+	RouteNamespaceSelector *metav1.LabelSelector `json:"routeNamespaceSelector" protobuf:"bytes,2,opt,name=routeNamespaceSelector"`
```

Comment:
Change this to `AllowedRouteNamespaces`. Should stay a pointer, because we want the nil behaviour. Should have `omitempty` added to JSON tags.

Reply:
Importantly, in this case, `omitempty` means "omit if nil", not "omit if the label selector is empty". So an empty label selector will *not* be omitted on output.

Reply:
Updated, thanks!